frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BranchCreationUtils.java

```diff
@@ -0,0 +1,48 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.GitLocalBranch;
+import git4idea.repo.GitRepository;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitMacheteBundle.class)
+public final class BranchCreationUtils {
+
+  private BranchCreationUtils() {}
+
+  @UIThreadUnsafe
+  public static @Nullable GitLocalBranch findLocalBranch(GitRepository gitRepository, String branchName) {
+    return gitRepository.getBranches().findLocalBranch(branchName);
+  }
+
+  @UIThreadUnsafe
+  public static void waitForCreationOfLocalBranch(GitRepository gitRepository, String branchName) {
```

Comment:
Hmmm just to make the API more clear... I'd return a boolean from this method (whether the branch has been created). While it might not be used for anything now... it makes it clear that in case the branch has **not** been created, no exception is thrown. This might be important for certain usages of this method, i.e. where subsequent code assumes that if `waitForCreationOfLocalBranch` returned, then the branch always exists.